code/modules/projectiles/gun_system.dm

```diff
@@ -1619,8 +1635,12 @@
 
 	if(((flags_item & WIELDED) && wielded_stable()) || CHECK_BITFIELD(flags_item, IS_DEPLOYED) || (master_gun && CHECK_BITFIELD(master_gun.flags_item, WIELDED) && master_gun.wielded_stable()))
 		gun_accuracy_mult = accuracy_mult
+		scatter = max(max(min_scatter, 0), min((scatter + scatter_increase) - ((world.time - last_fired - 1) * scatter_decay), max_scatter))
 		gun_scatter = scatter
 		wielded_fire = TRUE
+	else
+		scatter_unwielded = max(max(min_scatter_unwielded, 0), min((scatter + scatter_increase_unwielded) - ((world.time - last_fired - 1) * scatter_decay_unwielded), max_scatter_unwielded))
+		gun_scatter = scatter_unwielded
```

Comment:
This is adding a floor to scatter of 0. As well as causing bugs like that bipod one, it also means that anything that effects gun_scatter (instead of scatter) will ensure a positive value, where as currently many guns have negative scatter for the purpose of offsetting those penalties.
i.e, movement, burst, staggered, etc.

Would it be better to have a separate var for 'current scatter' etc, instead of modifying scatter var directly?

Reply:
yeah that makes sense, the floor of 0 should be further down which already exists as well, also no need since if you dont setup the increase/decay or a min/max it shoulnd't change the scatter at all, it uses scatter because when the system is in work having scatter var as it was doenst make sense anymore

Reply:
pending?